auth-web/src/components/auth/staff/continuation-application/PreviousCorrespondence.vue

```diff
@@ -0,0 +1,185 @@
+import { computed, defineComponent, reactive } from '@vue/composition-api'
+import AutoResize from 'vue-auto-resize'
+import moment from 'moment-timezone'
+
+export default defineComponent({
+  name: 'PreviousCorrespondence',
+
+  directives: { AutoResize },
+
+  props: {
+    review: { type: Object as () => ContinuationReviewIF, required: true },
+    filing: { type: Object as () => ContinuationFilingIF, required: true }
+  },
+
+  setup (props) {
+    const state = reactive({
+      /** The list of correspondence items. */
+      correspondences: computed<Array<any>>(() => {
+        if (!props.review) return [] // safety check
+
+        const list = []
+
+        // always add initial submission
+        // get date and user from review object
+        list.push({
+          label: `${strToPacificDateTime(props.review.creationDate)} &mdash; ` +
+            `Application Submitted &mdash; ${props.review.completingParty}`,
+          body: ''
+        })
+
+        // add items according to review results
+        props.review.results.forEach((result) => {
+          // did staff request a change?
+          if (result.status === ReviewStatus.CHANGE_REQUESTED) {
+            list.push({
+              label: `${strToPacificDateTime(result.creationDate)} &mdash; ` +
+                `Change Requested &mdash; ${result.reviewer}`,
+              body: result.comments
+            })
+
+            // did user resubmit?
+            // get date from result object; don't display user
+            if (result.submissionDate) {
+              list.push({
+                label: `${strToPacificDateTime(result.submissionDate)} &mdash; ` +
+                  'Application Resubmitted',
+                body: ''
+              })
+            }
+          }
+
+          // did staff approve the review?
+          if (result.status === ReviewStatus.APPROVED) {
+            list.push({
+              label: `${strToPacificDateTime(result.creationDate)} &mdash; ` +
+                `Application Approved &mdash; ${result.reviewer}`,
+              body: result.comments
+            })
+          }
+
+          // did staff reject the review?
+          if (result.status === ReviewStatus.REJECTED) {
+            list.push({
+              label: `${strToPacificDateTime(result.creationDate)} &mdash; ` +
+                `Application Rejected &mdash; ${result.reviewer}`,
+              body: result.comments
+            })
+          }
+        })
+
+        return list
+      })
+    })
+
+    /**
+     * Converts a date-time string to a Pacific date-time string.
+     * Sample input: "2024-06-04T04:20:00+00:00".
+     * Sample output: "Jun 3, 2024 at 9:20 pm Pacific Time".
+     */
+    function strToPacificDateTime (str: string): string {
+      const date = moment.utc(str).toDate()
+      return (date) ? moment(date).tz('America/Vancouver').format('MMM D, YYYY [at] h:mm a [Pacific time]') : ''
+    }
+
+    return {
+      ...state
```

Comment:
...toRefs(state) ? not sure if it makes a diff

Reply:
No refs declared in this component.

Reply:
all good, either way works for me